plugin/src/test/groovy/org/gradle/testretry/AbstractPluginFuncTest.groovy

```diff
@@ -78,6 +78,24 @@ abstract class AbstractPluginFuncTest extends Specification {
                         throw new java.io.UncheckedIOException(e);
                     }
                 }
+
+                public static void flakyAssertPassFailPass(String id) {
+                    Path marker = Paths.get("build/marker.file." + id);
+                    try {
+                        if (Files.exists(marker)) {
+                            int counter = Integer.parseInt(new String(Files.readAllBytes(marker)));
+                            ++counter;
+                            Files.write(marker, Integer.toString(counter).getBytes());
```

Comment:
```suggestion
                            marker.text = counter
```

Reply:
❗ This is a Java class that we create in the build workspace to simplify creation of test classes, so this won't work.